src/hooks/useCreateUser.tsx

```diff
@@ -0,0 +1,49 @@
+// src/hooks/useCreateUser.ts
+import { useState } from 'react';
+import { API_PATH } from '@/utils/constants';
+import { User } from '@/interfaces/User'; // Import the User interface
+
+// Define the input type for creating a user, excluding fields that are auto-generated like created_at and updated_at.
+type CreateUserInput = Omit<User, 'user_id' | 'created_at' | 'updated_at'> & {
+  password: string; // Include password separately since it’s used for creating users
+};
+
+const useCreateUser = () => {
+  const [userId, setUserId] = useState<string | null>(null); // Only store the user ID
+  const [error, setError] = useState<string | null>(null);
+  const [loading, setLoading] = useState<boolean>(false);
+  const [success, setSuccess] = useState<string | null>(null);
```

Comment:
you can remove these and here is why:
when you actually do use the hook with useMutation, that built in hook already comes with loading, success, and error predefined. Later on these will be dead code once you do implement the hook. Good thinking with trying to catch these though